clusterloader2/testing/load/modules/measurements.yaml

```diff
@@ -158,6 +159,21 @@ steps:
       - name: Perc50
         query: quantile_over_time(0.50, sum by (container) (container_memory_working_set_bytes / 1024 / 1024)[%v:])
 {{end}}
+{{if $ENABLE_WATCH_CLOSED_MEASUREMENT}}
+  - Identifier: WatcherPerformanceMetrics
+    Method: GenericPrometheusQuery
+    Params:
+      action: {{$action}}
+      metricName: Watcher Performance
+      metricVersion: v1
+      queries:
```

Comment:
There's no such metric as `terminated_watchers_total` but rather `apiserver_watch_cache_terminated_watchers_total` - we prepend the metric name with the `System`'s name here: https://github.com/kubernetes/kubernetes/blob/8e48df135318026d5f8a972a96a2ff665889568a/staging/src/k8s.io/apiserver/pkg/storage/cacher/metrics/metrics.go#L90.

In general, I'd like to see two queries here:

* `sum(increase(apiserver_terminated_watchers_total[%v:])) by (resource)` - this will inform us about the number of watch connections being forcefully terminated by the kube-apiserver during test.
* `sum(increase(apiserver_watch_cache_initializations_total[%v:])) by (resource)` - this will inform us if any of the watch cache's watches had to reinitialize during test.

Given that, please also add the `dimensions` field within the `Params` field so it contains the `resource` dimension.

Reply:
fixed